__tests__/app.test.js

```diff
@@ -195,3 +195,67 @@ describe('GET /api/articles/:article_id/comments', () => {
       });
   });
 });
+describe('POST /api/articles/:article_id/comments', () => {
+  test.skip('201: successfully adds a comment for an article and responds with the posted comment', () => {
+    const newComment = {
+      author: 'albertthelad',
+      body: 'Great article! Needs more biscuits though...',
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            article_id: 1,
+            author: 'albertthelad',
+            body: 'Great article! Needs more biscuits though...',
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+
+  test('400: responds with "Bad Request" if author is missing in request body', () => {
+    const newComment = {
+      body: 'Mmm delicious',
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+
+  test('400: responds with "Bad Request" if body is missing in the request body', () => {
+    const newComment = {
+      author: 'opaltheeeper123',
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+
+  test('400: responds with "Bad Request" if body is too short (less than 1 character)', () => {
+    const newComment = {
+      author: 'theWalkingDictionary',
+      body: '',
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+});
```

Comment:
We should also test that the if author of the comment _isn't found_ in the users table we get a suitable error code back. PSQL will be your friend for this circumstance - console the error in the catch to see how to proceed when running the test

Reply:
We also need to handle if the id is valid but nonexistent eg `/api/articles/99999/comments` as well as if the id is invalid eg `/api/not_an_id/comments`